src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs

```diff
@@ -77,10 +77,10 @@ private static IServiceCollection AddSignalRServiceCore(this IServiceCollection
         {
             services.AddSingleton<IServiceManager, ServiceManagerImpl>();
             services.PostConfigure<ServiceManagerOptions>(o => o.ValidateOptions());
-            services.TryAddSingleton<IEndpointRouter, AutoHealthCheckRouter>();
             var tempServices = new ServiceCollection().AddSignalR()
                 .AddAzureSignalR<CascadeServiceOptionsSetup>().Services;
             services.Add(tempServices.Where(service => service.ServiceType != typeof(IHostedService)));
+            services.AddSingleton<IEndpointRouter, AutoHealthCheckRouter>();
```

Comment:
this actually add two IEndpointRouter to the service? 

`AddAzureSignalR` is actually calling `TryAddSingleton`, so you can also try 
`new services => services.Add the one you want => services.AddAzureSignalR..... =>...`